executor/sort.go

```diff
@@ -208,35 +248,26 @@ func (e *SortExec) fetchRowChunks(ctx context.Context) error {
 		if rowCount == 0 {
 			break
 		}
-		if e.alreadySpilled() {
-			// append chk to disk.
-			err := e.rowChunksInDisk.Add(chk)
-			if err != nil {
-				return err
-			}
-		} else {
-			e.rowChunks.Add(chk)
-			if atomic.LoadUint32(&e.exceeded) == 1 {
-				e.rowChunksInDisk, err = e.spillToDisk()
-				if err != nil {
-					return err
-				}
-				e.memTracker.Consume(-e.rowChunks.GetMemTracker().BytesConsumed())
-				e.rowChunks = nil // GC its internal chunks.
-				atomic.StoreUint32(&e.spilled, 1)
-			}
+		if err := e.rowChunks.Add(chk); err != nil {
+			return err
 		}
+		if e.rowChunks.AlreadySpilled() {
+			e.rowChunks = chunk.NewRowContainer(retTypes(e), e.maxChunkSize)
+			e.rowChunks.GetMemTracker().AttachTo(e.memTracker)
+			e.rowChunks.GetMemTracker().SetLabel(rowChunksLabel)
+			e.rowChunks.SetOnExceededCallback(onExceededCallback)
+			e.spillAction.SetRowContainer(e.rowChunks)
+			e.spillAction.ResetOnce()
+		}
+	}
+	if e.rowChunks.NumRow() > 0 && len(e.partitionList) > 0 {
```

Comment:
```suggestion
	if e.rowChunks.NumRow() > 0 {
```
We could sort the last `rowChunks` here.